inst/tests/tests.Rraw

```diff
@@ -993,16 +993,17 @@ test(343, DT[,(mynewcol):=21L], data.table(b=1:3,newname=21L))
 mycols = 1:2
 test(344, DT[,(mycols):=NULL], data.table(NULL))
 
-
-# It seems that the .Internal rbind of two data.frame coerces IDate to numeric. Tried defining
```

Comment:
Wow, the level of _deja vu_ on reading this:

> It seems that the .Internal rbind of two data.frame coerces IDate to numeric. Tried defining "[<-.IDate" as per Tom's suggestion, and c.IDate to no avail (maybe because the .Internal code in bind.c doesn't look up package methods?)

In fact I went through the same existential crisis for a while this morning! I'm not sure what was tried for `[<-.IDate` back then, maybe the `NextMethod` approach? I tried that as well but it seems fundamentally doomed because `...` in `NextMethod` can't be passed as `origin` to the `as.Date` call in `[<-.Date`, which is needed if we pass the underlying `integer` to `[<-.Date` as `value`. If we pass an `IDate`, the #2008 problem happens -- `double` with `IDate` class. I didn't try running `as.Date(value)` but that would mean coercing numeric just to get `[<-.Date` and then back to integer aftewards... seems silly.

Anyway I think we can change this test that was testing that a bug was still a bug. Now the bug is fixed. I'm not sure if the subsequent test is still needed, but it can't hurt.